src/App.test.jsx

```diff
@@ -1,20 +1,69 @@
+import App from './App.jsx';
 import {
   render,
   screen,
   waitForElementToBeRemoved,
 } from '@testing-library/react';
-
-import App from './App.jsx';
+import userEvent from '@testing-library/user-event';
+import { MemoryRouter, Route } from 'react-router';
+import { act } from 'react-dom/test-utils';
 
 test('making sure the header renders and a list of pokemon who up ion loading', async () => {
```

Comment:
I would clean up this test file a little. Also combining all of these tests together will solve your memory leak issue. Because this is all technically testing the functionality of one app it probably makes more sense to have a single test.
https://kentcdodds.com/blog/write-fewer-longer-tests